### ISSUE TITLE
[Bug]: Error when listing users if username `admin` doesn't exist

### ISSUE THREAD

**Issue by theoludwig** [nextcloud/server#39618]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Currently it is not possible to change/rename login name as explained in the docs: https://docs.nextcloud.com/server/latest/admin_manual/configuration_user/user_configuration.html#renaming-a-user and  it is a known issue: #5488.

In my use case, I'm the only user using my Nextcloud instance, and would rather avoid using `admin` as login name, but instead use my name (e.g: `theoludwig`).
As it is impossible to rename the `admin` username, I found this reply explaining a workaround: https://help.nextcloud.com/t/how-to-change-admin-username-in-nc-9-0-54/5540/2 (create a new user, add him to admin group, and delete old admin account), that work perfectly!

However in the `/settings/users` page that list all users, when the user `admin` doesn't exist, it displays an error: "An error occurred during the request. Unable to proceed.".

![image](https://github.com/nextcloud/server/assets/25207499/9dda7520-c73a-4123-b401-677fd579d26e)

There is this error because when we go to this page, it does a HTTP GET Request to `/ocs/v2.php/cloud/users/admin`, and that returns a 404 error with this reponse:

```json
{
    "ocs": {
        "meta": {
            "status": "failure",
            "statuscode": 404,
            "message": "User does not exist"
        },
        "data": []
    }
}
```

That make sense as the `admin` user doesn't exist.

### Steps to reproduce

1. Create a new user, and add him to the admin group.
2. Delete the orginal first created user `admin`.
3. Go to the `/settings/users` page.
4. Notice the error message.

### Expected behavior

This is not a critical bug, and is only annoying. But it would make sense to not automatically do the request for the `admin` user, when it's not needed. This would also improve performance to avoid doing an extra not needed/useless request.

### Installation method

Official All-in-One appliance

### Nextcloud Server version

27.0.1

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

PostgreSQL

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by szaimen** [comment]
Cc @nextcloud/server

**Comment by solracsf** [comment]
I have no such user `admin` on my instance and i can list user.
But, i have never created an user named `admin` in the first place.

**Comment by joshtrichards** [comment]
Same as @solracsf. Nor can I reproduce this issue on a current(ish) `master` test instance that does (well, did) have an `admin` account.

I do note that Step 2 of the reproduction steps doesn't specify whether one logs out of the to-be-deleted account first or not. I certainly considered that wise so I logged out of `admin` and logged into the new replacement account `newadmin` in order to do the actual deletion of `admin`. Any chance you were still logged in as `admin` @theoludwig?

**Comment by theoludwig** [comment]
Thank you for your investigations!

I tried reproducing it using VSCode devcontainers on `stable27` branch, but everything work fine, couldn't reproduce.
I was not logged in as `admin`, and deleted the admin account using the new user account `theoludwig`.

I don't know what caused the issue, but the error message keeps appearing on my production Nextcloud instance.
Is there is any way, I can debug this? Is that an issue with [Nextcloud AIO](https://github.com/nextcloud/all-in-one)?
Also I tried recreating a `admin` user (also this user doesn't need to be in the `admin` group/to be a admin, only the username matters), and the error message disappear when refreshing the page `/settings/users`. If I delete him again, same error.
The current workaround, is to have a `admin` account but disabled, so no one can sign in to this account, and that way there isn't this error message, but I would rather find a way to completely delete this `admin` user.
